dask/dataframe/core.py

```diff
@@ -629,12 +647,23 @@ def _loc(df, start, stop, include_right_boundary=True):
     2  20
     2  30
     """
-    result = df.loc[slice(start, stop)]
+    result = df.loc[start:stop]
     if not include_right_boundary:
         # result = df[df.index != stop]
         result = result.iloc[:result.index.get_slice_bound(stop, 'left',
                                                    result.index.inferred_type)]
-    return result
+    if not result.empty:
+        return result
+
+    # this is horrible
+    # i need to find a non-type-specific way to generate a range containing
+    # values that don't exist in the index and fill that with NaNs
+    d = np.diff(df.index.values)
```

Comment:
i think a new `repartition` ish function is probably in order since i don't necessarily want to have empty datetime indices *always* perform a reindex style operation